sdks/java/core/src/main/java/org/apache/beam/sdk/util/WindowedValue.java

```diff
@@ -291,23 +332,23 @@ public ValueInEmptyWindows(T value, PaneInfo pane) {
     public boolean equals(Object o) {
       if (o instanceof ValueInEmptyWindows) {
         ValueInEmptyWindows<?> that = (ValueInEmptyWindows<?>) o;
-        return Objects.equals(that.pane, this.pane)
-            && Objects.equals(that.value, this.value);
+        return Objects.equals(that.getPane(), this.getPane())
```

Comment:
I strongly doubt this was empirically determined. Such micro-optimization and should be commented in detail if we  reproduce it.